src/main/java/com/onarandombox/MultiverseCore/destination/DestinationFactory.java

```diff
@@ -37,19 +43,103 @@ public DestinationFactory(MultiverseCore plugin) {
     }
 
     /**
-     * Gets a new destination from a string.
+     * Gets a new destination from a string and possibly for a player.
      * Returns a new InvalidDestination if the string could not be parsed.
      *
      * @param destination The destination in string format.
+     * @param player The player.
      *
      * @return A non-null MVDestination
      */
-    public MVDestination getDestination(String destination) {
+    public MVDestination getDestination(String destination, Player player) {
```

Comment:
Probably better you leave the old method in just add this as a new method. This way we can preserve backwards compatibility.